bin/dfx-neuron-create

```diff
@@ -7,56 +7,82 @@ source "$SOURCE_DIR/clap.bash"
 # Define options
 clap.define short=s long=icp desc="The amount of ICP to icp" variable=ICP default=1
 clap.define short=i long=DFX_IDENTITY desc="The dfx identity to use" variable=DFX_IDENTITY default="$(dfx identity whoami)"
+clap.define long=retries desc="The number of retries" variable=DFX_RETRIES default="20"
+clap.define long=retry_interval desc="The number of seconds between retries" variable=DFX_RETRY_SLEEP default="4"
 clap.define short=n long=network desc="The dfx network to use" variable=DFX_NETWORK default="local"
 clap.define short=V long=verbose desc="Print progress" variable=DFX_VERBOSE nargs=0
 # Source the output file ----------------------------------------------------------
 source "$(clap.build)"
 test -z "${DFX_VERBOSE:-}" || set -x
 
-# Create a neuron
-PEM="$HOME/.config/dfx/identity/$DFX_IDENTITY/identity.pem"
-test -e "$PEM" || {
-  echo "ERROR: Pem file for user '$DFX_IDENTITY' does not exist at '$PEM'"
-  exit 1
-} >&2
-NEURONS_DIR="$HOME/.config/dfx/identity/$DFX_IDENTITY/neurons"
-mkdir -p "$NEURONS_DIR"
-NEURONS_FILE="$NEURONS_DIR/$DFX_NETWORK"
-IC_URL="$("$SOURCE_DIR/dfx-network-url" --network "$DFX_NETWORK")"
-export IC_URL
-# Note: Quill send puts a pile of junk on stdout and has no option to put just the response(s) somewhere safe such as a file.
+retry() {
+  (
+    set -eu
+    for ((i = 0; i < DFX_RETRIES; i++)); do
+      if "${@}"; then
+        return
+      else
+        sleep "${DFX_RETRY_SLEEP}"
+      fi
+    done
+    {
+      echo "ERROR: Failed to run: $*"
+      exit 1
+    } >&2
+  )
+}
 
-command=(quill neuron-stake --insecure-local-dev-mode --pem-file "$PEM" --amount "$ICP" --name 1)
-MESSAGE="$("${command[@]}")"
-test -n "${MESSAGE:-}" || {
-  echo "ERROR: Created an empty message."
-  command -v quill
-  echo "${command[*]}"
-  exit 1
-} >&2
-command=(quill send --yes --insecure-local-dev-mode -)
-RESPONSE="$(echo "$MESSAGE" | "${command[@]}")"
-test -n "${RESPONSE:-}" || {
-  echo "ERROR: Received an empty response."
-  echo message:
-  echo "$MESSAGE"
-  echo
-  echo command -v quill
-  echo "${command[*]}"
-  exit 1
-} >&2
-NEURON_TEXT="$(echo "$RESPONSE" | perl -e 'while($_ = <>){if (s/NeuronId = record \{ id = ([0-9_]+) : nat64 \}/\1/){ print $_ }}')"
-test -n "${NEURON_TEXT:-}" || {
-  echo "ERROR: Could not find neuron in response:"
-  echo "$RESPONSE"
-  exit 1
-} >&2
-NEURON_ID="$(echo "$NEURON_TEXT" | tr -cd '[:alnum:]')"
-test -n "${NEURON_ID:-}" || {
-  echo "Failed to get neuron ID from line: $NEURON_TEXT"
-  exit 1
-} >&2
+mk_neuron() {
+  (
+    set -euo pipefail
```

Comment:
Does this have any effect?
When I tried adding this, it did *not* have the effect I wanted, which is:
1. When something inside `( ... )` fails, it doesn't continue running the block.
2. For the `( ... )` block to cause the outer script to fail when something inside it fails.

I tried this test script:
```#!/usr/bin/env bash
set -euo pipefail

DFX_RETRIES=5
DFX_RETRY_SLEEP=1

retry() {
  (
    set -eu
    for ((i = 0; i < DFX_RETRIES; i++)); do
      if "${@}"; then
        return
      else
        sleep "${DFX_RETRY_SLEEP}"
      fi
    done
    {
      echo "ERROR: Failed to run: $*"
      exit 1
    } >&2
  )
}

failing_func() {
  (
    set -euo pipefail
    echo before
    false
    echo after
  )
}

retry failing_func
```

And for me it outputs:
```
before
after
```

To get the intended effect, I have to replace `false` with `exit 1`.

Reply:
Setting euo pipefail is just habit.  It's a good question, whether this does any good here.  Below the code does call exit explicitly when a required step fails, so I do believe the code works as intended.

Reply:
On line 47:
```
IC_URL="$("$SOURCE_DIR/dfx-network-url" --network "$DFX_NETWORK")"
```
Previously, if the call to `dfx-network-url` would fail, the entire script would fail.
But now, if line 47 fails, it just continues with line 48.
This is surprising enough given the `set -euo pipefail` on line 2 but even more surprising given the `set -euo pipefail` on line 37.

If you want to be sure the script works the same as before, you'd have to replace line 47 with something like
```
if ! IC_URL="$("$SOURCE_DIR/dfx-network-url" --network "$DFX_NETWORK")"; then
  exit 1
fi
```
And similar for many other lines.
I'm not sure that's an improvement, but you should at least acknowledge it (perhaps with a comment on line 36) and it would be a good idea to go through the script and confirm that it fails when it should.